src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -221,70 +223,56 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
-            this.props.store.coverageInformation.status === "error";
-
-        const loadingDownloadData = loadingGeneAlterationData ||
-            this.props.store.nonOqlFilteredCaseAggregatedData.status === "pending";
-
-        const errorDownloadData = errorCaseAlterationData ||
-            this.props.store.nonOqlFilteredCaseAggregatedData.status === "error";
-
-        return (
-            <WindowWidthBox data-test="downloadTabDiv" offset={60}>
-                <div className={"tabMessageContainer"}>
-                    <OqlStatusBanner className="download-oql-status-banner" store={this.props.store} tabReflectsOql={true} />
-                </div>
-                <div>
-                    <FeatureTitle
-                        title="Downloadable Data Files"
-                        className="forceHeaderStyle h4"
-                        isLoading={this.downloadableFilesTable.isPending}
-                        style={{marginBottom:15}}
-                    />
-                    {this.downloadableFilesTable.isComplete && this.downloadableFilesTable.result}
-                </div>
-                <hr/>
-                <div className={styles["tables-container"]} data-test="dataDownloadGeneAlterationTable">
-                    <FeatureTitle
-                        title="Gene Alteration Frequency"
-                        isLoading={this.geneAlterationData.isPending}
-                        className="pull-left forceHeaderStyle h4"
-                    />
-                    {this.geneAlterationData.isComplete && (<GeneAlterationTable geneAlterationData={this.geneAlterationData.result} />)}
-                </div>
-                <hr/>
-                <div className={styles["tables-container"]}>
-                    <FeatureTitle
-                        title="Type of Genetic Alterations Across All Samples"
-                        isLoading={this.caseAlterationData.isPending || this.oqls.isPending}
-                        className="pull-left forceHeaderStyle h4"
-                    />
-                    {this.oqls.isComplete && this.caseAlterationData.isComplete && this.props.store.alterationsBySelectedMolecularProfiles.isComplete && (
-                        <CaseAlterationTable
-                            caseAlterationData={this.caseAlterationData.result}
-                            oqls={this.oqls.result}
-                            alterationTypes={this.props.store.alterationsBySelectedMolecularProfiles.result}
-                        />
-                    )}
-                </div>
-            </WindowWidthBox>
-        );
+        const status = getMobxPromiseGroupStatus((this.downloadableFilesTable, this.geneAlterationData, this.caseAlterationData, this.oqls));
+
+        switch (status) {
+            case "pending":
+                return <LoadingIndicator isLoading={true} center={true} size={"big"}/>;
+            case "error":
+                return <ErrorMessage/>;
```

Comment:
This looks simpler but now, for example, a single error for downloadable files table will prevent rendering of error-free gene alteration and case alteration tables. I don't know if it is a big deal though.

Reply:
Thanks Onur, that's a good point. @alisman Do we still want to show the rest of good tables if we got the error.